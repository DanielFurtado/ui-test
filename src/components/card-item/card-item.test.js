import React, { useState as useStateMock } from 'react';
import CardItem from './card-item.component';
import userEvent from '@testing-library/user-event';
import {render, screen} from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

const data = {
  'user': {
    'name': 'Daniel'
  }
};

describe(('Card Item'), () => {
  it('matches the snapshot', () => {
    const component = renderer.create(<CardItem image={data}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the user name', () => {
    const {getByText} = render(<CardItem image={data} />);
    expect(getByText('Daniel')).toBeInTheDocument();
  });
  
  it('handles click correctly', async () => {
    const {getByText} = render(<CardItem image={data} />);
    const user = userEvent.setup();  
    await user.click(screen.getByTestId('card-container'));
    expect(getByText('Preview')).toBeInTheDocument();
  });
});
