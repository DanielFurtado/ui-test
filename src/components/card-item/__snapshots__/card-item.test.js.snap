// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item matches the snapshot 1`] = `
<div
  className="card-container"
  data-testid="card-container"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(undefined.jpg)",
    }
  }
>
  <div
    className="card-body-container"
  >
    <div
      className="card-info shine"
    >
      <p>
        Daniel
      </p>
      <span
        className="arrow-right"
      />
    </div>
  </div>
</div>
`;
